### DISCUSSION TITLE
API vs WebInterface - URI parameters not passed the same

### DISCUSSION THREAD

**Discussion by GrepZen** [YOURLS/YOURLS#3818] [Q&A] [answered]
Sample_LongURL:  [ http://ExampleDomain.com?Parameter0=Value0&Parameter1=Value1&Parameter2=Value2 ]

**Method_WebInterface:** If I enter the 'Sample_LongURL' into the web interface, YOURLS will create a working 'ShortURL' that will, when browsed to carry over all of the URI parameters (Parameter0, Parameter1, Parameter2 ...etc)  

**METHOD_API:**  However, If I use the API(yourls-api.php) like:
http://MyDomain.com/yourls-api.php?signature=SOMEVALIDSIGNATURE&action=shorturl&format=simple&url=http://ExampleDomain.com?Parameter0=Value0&Parameter1=Value1&Parameter2=Value2   
Then, the returned ShortURL will redirect to an incorrect URL of: http://ExampleDomain.com?Parameter0=Value0  
It cuts off every thing AFTER the first URI parameter.  

I have also tried PlugIn(yourls-plugin-forward-query-strings-main)  ==NO JOY

Is anyone able to replicate this behavior? 
//Or perhaps recommend a fix?

**Comment by dgw** [comment, marked as answer]
You must encode special characters inside a GET parameter value. See #3010.

Depending how you are calling the API, you might have functionality available that will handle this for you transparently. For example, in the `requests` library for Python, [passing the API options as `params`](https://requests.readthedocs.io/en/latest/user/quickstart/#passing-parameters-in-urls) takes care of this.

**Reply by GrepZen** [reply]
I've URL_ENCODED the URL string and it works perfectly!

@dgw MUCH appreciated. Thank You!
